### PR TITLE
Cleanup READMEs.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

I'd like to make this repo something other plugin repos can inspire themselves from in terms of standard content.

- Added ADMINS.md documenting admin responsibilities.
- Standardized CONTRIBUTING.md
- Added DEVELOPER_GUIDE.md and moved content from README.
- Linked standard MAINTAINERS.md.
- Added SECURITY.md.
- Cleaned up README.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List

- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
@CEHENKLE and @saratvemulapalli I'd like to use this repo as a sample for other plugin repos in terms of required content. Please take a close look!

File: README.md [hunk 1]
```diff
@@ -1,58 +1,47 @@
...
-
-See the [LICENSE](./LICENSE.txt) file for our project's licensing. We will ask you to confirm the licensing of your contribution.
-
+This project is licensed under the [Apache v2.0 License](LICENSE.txt).
 
 ## Copyright
 
-Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
\ No newline at end of file
+Copyright 2020-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (saratvemulapalli) [inline README.md:47]: Just curious:
1. Why do we still have 2020? Our project started in 2021...
2. How should be take care of updating the new year. Eg 2022

**Review by saratvemulapalli — APPROVED** [review]
Looks good to me.
Thanks dB!

File: README.md (hunk 1 above)
Comment (dblock) [reply README.md:47]: I assume this comes from OpenDistro? @elfisher maybe you know?

**Pull request merged**